udp_all.go

```diff
@@ -33,14 +33,26 @@ func (ua *udpAddr) Equals(t *udpAddr) bool {
 }
 
 func (ua *udpAddr) String() string {
+	if ua == nil {
+		return ""
+	}
```

Comment:
```suggestion
	if ua == nil {
		return "<nil>"
	}
```

I think we should match what the stdlib does for similar things here, so it isn't confusing having a blank string in logs for this error case: https://github.com/golang/go/blob/go1.16.3/src/net/ip.go#L294-L296